Assets/Features/Building/Scripts/Grid/Cursor.cs

```diff
@@ -0,0 +1,367 @@
+        {
+            //Sets the offset of the whole grid
+            Vector3Int offset = new Vector3Int(Mathf.RoundToInt(_tilemap.transform.position.x),
+                Mathf.RoundToInt(_tilemap.transform.position.y),
+                Mathf.RoundToInt(_tilemap.transform.position.z));
+
+            //get the current selection of tiles
+            List<SubBuildItem> currentSelectedGroup = new List<SubBuildItem>();
+            foreach (var shape in shapes)
+            {
+                currentSelectedGroup.Add(new SubBuildItem(shape.Tile, position + shape.GridPosition));
+            }
+
+            //remove the tiles that were in the previous selection, but not in the current
+            foreach (var gridPosition in _selectedGroup)
+            {
+                if (!currentSelectedGroup.Any(x => x.GridPosition == gridPosition.GridPosition))
+                {
+                    _tilemap.SetTile(gridPosition.GridPosition - offset, null);
+                }
+            }
+
+            //check if all positions are required when placing
+            bool valid = true;
+            if (requireAllAvailable)
+            {
+                foreach (var gridPosition in currentSelectedGroup)
+                {
+                    if (!_grid.IsEmpty(gridPosition.GridPosition))
+                    {
+                        valid = false;
+                        break;
+                    }
+                }
+            }
+
+            var validColor = !flipColors ? _validColor : _invalidColor;
+            var invalidColor = !flipColors ? _invalidColor : _validColor;
+            
+            //set the tile on the tilemap
+            foreach (var gridPosition in currentSelectedGroup)
+            {
+                TileChangeData tempTile = new TileChangeData()
+                {
+                    position = gridPosition.GridPosition - offset,
+                    color = valid
+                        ? (_grid.IsEmpty(gridPosition.GridPosition) ? validColor : invalidColor)
+                        : invalidColor
+                };
+
+                tempTile.transform = Matrix4x4.Rotate(Quaternion.Euler(0, 0, _rotation * -90));
+                tempTile.tile = gridPosition.Tile;
+                _tilemap.SetTile(tempTile, true);
+            }
+
+            _selectedGroup = currentSelectedGroup;
+        }
+
+        public void ChangeToBuilding(BuildableObject buildableObject)
+        {
+            _rotation = 0;
+            _selectedBuilding = buildableObject;
+
+            //Create a fresh shape so the rotation is correct and doesn't change the prefab
+            _shape = new List<SubBuildItem>();
+            foreach (var item in buildableObject.BuildItems)
+            {
+                _shape.Add(new SubBuildItem(item.Tile,
+                    new GridPosition(item.GridPosition.Position, item.GridPosition.Layer)));
+            }
+        }
+
+        public void DisableCursor()
```

Comment:
Not sure why disabling the cursor hides the tilemap instead. Same for enabling the cursor

Reply:
Seperate Cursor tilemap